src/test-harness/SparkEthereumTests.sol

```diff
@@ -529,50 +550,107 @@ abstract contract SparkEthereumTests is SparklendTests {
 
-        // Check spreads
-        assertEq(IRateTargetBaseIRM(oldIRM).getBaseVariableBorrowRateSpread(), oldSpread);
-        assertEq(IRateTargetBaseIRM(newIRM).getBaseVariableBorrowRateSpread(), newSpread);
+        uint256 ssrRate = uint256(IRateSource(ICustomIRM(newParams.irm).RATE_SOURCE()).getAPR());
 
-        // Check change in base variable borrow rate
-        assertEq(IDefaultInterestRateStrategy(oldIRM).getBaseVariableBorrowRate(), oldBaseRate);
-        assertEq(IDefaultInterestRateStrategy(newIRM).getBaseVariableBorrowRate(), newBaseRate);
+        ReserveConfig memory configBefore = _findReserveConfigBySymbol(createConfigurationSnapshot('', ctx.pool), symbol);
+
+        _validateInterestRateStrategy(
+            configBefore.interestRateStrategy,
+            oldParams.irm,
+            InterestStrategyValues({
+                addressesProvider:             address(ctx.poolAddressesProvider),
+                optimalUsageRatio:             oldParams.optimalUsageRatio,
+                optimalStableToTotalDebtRatio: 0,
+                baseStableBorrowRate:          oldParams.variableRateSlope1,
+                stableRateSlope1:              0,
+                stableRateSlope2:              0,
+                baseVariableBorrowRate:        ssrRate + oldParams.baseRateSpread,
+                variableRateSlope1:            oldParams.variableRateSlope1,
+                variableRateSlope2:            oldParams.variableRateSlope2
+            })
+        );
+
+        executeAllPayloadsAndBridges();
+
+        ReserveConfig memory configAfter = _findReserveConfigBySymbol(createConfigurationSnapshot('', ctx.pool), symbol);
+
+        _validateInterestRateStrategy(
+            configAfter.interestRateStrategy,
+            newParams.irm,
+            InterestStrategyValues({
+                addressesProvider:             address(ctx.poolAddressesProvider),
+                optimalUsageRatio:             newParams.optimalUsageRatio,
+                optimalStableToTotalDebtRatio: 0,
+                baseStableBorrowRate:          newParams.variableRateSlope1,
+                stableRateSlope1:              0,
+                stableRateSlope2:              0,
+                baseVariableBorrowRate:        ssrRate + newParams.baseRateSpread,
+                variableRateSlope1:            newParams.variableRateSlope1,
+                variableRateSlope2:            newParams.variableRateSlope2
+            })
+        );
+    }
+
+    function _testRateTargetKinkIRMUpdate(
+        string                  memory symbol,
+        RateTargetKinkIRMParams memory oldParams,
+        RateTargetKinkIRMParams memory newParams
+    )
+        internal
+    {
+        SparkLendContext memory ctx = _getSparkLendContext();
+
+        // Rate source should be the same
+        assertEq(ICustomIRM(newParams.irm).RATE_SOURCE(), ICustomIRM(oldParams.irm).RATE_SOURCE());
```

Comment:
Same here I would specifically validate the kink spread parameter.

Reply:
Added assertions